client/blocks/reader-export-button/index.jsx

```diff
@@ -43,29 +43,41 @@ class ReaderExportButton extends React.Component {
 
 	state = { disabled: false };
 
-	onClick = () => {
+	isMounted = false;
+
+	componentDidMount() {
+		this.isMounted = true;
+	}
+
+	componentWillUnmount() {
+		this.isMounted = false;
+	}
+
+	onClick = async () => {
 		// Don't kick off a new export request if there's one in progress
-		if ( this.props.disabled || this.state.disabled ) {
+		if ( ! this.isMounted || this.props.disabled || this.state.disabled ) {
 			return;
 		}
 
+		this.setState( {
+			disabled: true,
+		} );
+
+		let data;
+
 		if ( this.props.exportType === READER_EXPORT_TYPE_LIST ) {
-			wpcom.undocumented().exportReaderList( this.props.listId, this.onApiResponse );
+			data = await wp.req.get( `/read/lists/${ this.props.listId }/export`, {
```

Comment:
Ideally, we should have error handling for those, but since we didn't have any before, we're good to go.

Reply:
I notice the old code did check for an error in the API response, so I've added a `try...catch` to do the same.

Let me know if there's a more elegant way to handle it!

Reply:
Sounds good. A `try / catch` is how we usually do it anyway 👍